### PR TITLE
#18390 Fix Exasol VS display for SaaS.

### LINKED ISSUE #18390: Expanding "Virtual Schemas" under a non-privileged user in Exasol leads to error for Exasol SaaS DB
### Description

At the moment expanding "Virtual Schemas" under a non-privileged user in Exasol leads to error for Exasol SaaS DB:

```
SQL Error [42000]: object ADAPTER_SCRIPT not found [line 1, column 122] (Session: 1750920676951195651)
  object ADAPTER_SCRIPT not found [line 1, column 122] (Session: 1750920676951195651)
```

### DBeaver Version

Community Edition 22.2.5.202211202223

### Operating System

Windows

### Database and driver

Exasol JDBC driver 7.1.11

### Steps to reproduce

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by allipatev**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.exasol/src/org/jkiss/dbeaver/ext/exasol/model/ExasolCurrentUserPrivileges.java [hunk 1]
```diff
@@ -62,6 +62,10 @@ public Boolean getatLeastV5() {
     public Boolean getatLeastV6() {
         return majorVersion >= 6;
     }
+
+    public Boolean getatLeastV8() {
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.exasol/src/org/jkiss/dbeaver/ext/exasol/model/ExasolCurrentUserPrivileges.java]:
```suggestion
    public Boolean getAtLeastV8() {
```

Comment (allipatev) [reply plugins/org.jkiss.dbeaver.ext.exasol/src/org/jkiss/dbeaver/ext/exasol/model/ExasolCurrentUserPrivileges.java]: Thanks @E1izabeth , should be fixed now.

**Review by E1izabeth — APPROVED** [review]

**Review by ShadelessFox — APPROVED** [review]

**Comment by LonwoLonwo** [comment]
Hello @allipatev 

Can you please provide your script for the condition:

this.exasolCurrentUserPrivileges.getAtLeastV8()

Too many escapings here. I'm confused.

File: plugins/org.jkiss.dbeaver.ext.exasol/src/org/jkiss/dbeaver/ext/exasol/model/ExasolDataSource.java [hunk 2]
```diff
@@ -137,7 +137,7 @@ public void initialize(@NotNull DBRProgressMonitor monitor)
 					"	s.SCHEMA_NAME as OBJECT_NAME," + 
 					"	s.SCHEMA_OWNER AS OWNER," + 
 					"CAST(NULL AS TIMESTAMP) AS created, " +
-					"	ADAPTER_SCRIPT," + 
+					"	" + (this.exasolCurrentUserPrivileges.getAtLeastV8() ? "'\"' || ADAPTER_SCRIPT_SCHEMA || '\".\"' || ADAPTER_SCRIPT_NAME || '\"' AS ADAPTER_SCRIPT" : "ADAPTER_SCRIPT") + "," +
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.exasol/src/org/jkiss/dbeaver/ext/exasol/model/ExasolDataSource.java]: " " - what for it?
Also, Line is longer than 140 characters (found 222), and Line contains a tab character - this is for our codacy

**Comment by allipatev** [comment]
hi @LonwoLonwo 
this.exasolCurrentUserPrivileges.getAtLeastV8() is defined in https://github.com/dbeaver/dbeaver/blob/f8119069690b8b3651b51a78a378077a816e2964/plugins/org.jkiss.dbeaver.ext.exasol/src/org/jkiss/dbeaver/ext/exasol/model/ExasolCurrentUserPrivileges.java#L66-L68

File: plugins/org.jkiss.dbeaver.ext.exasol/src/org/jkiss/dbeaver/ext/exasol/model/ExasolDataSource.java (hunk 2 above)
Comment (allipatev) [reply plugins/org.jkiss.dbeaver.ext.exasol/src/org/jkiss/dbeaver/ext/exasol/model/ExasolDataSource.java]: > " " - what for it?

To preserve the current formatting of the generated SQL query.

> Line is longer than 140 characters (found 222)

I'll try to split the line.

> Line contains a tab character - this is for our codacy

The added code is copy-paste, the file is already full of tab characters. Does it make sense to have alternating code style inside of a single file?

Comment (allipatev) [reply plugins/org.jkiss.dbeaver.ext.exasol/src/org/jkiss/dbeaver/ext/exasol/model/ExasolDataSource.java]: @LonwoLonwo 
Line length should be fixed now.

**Comment by LonwoLonwo** [comment]
Ok, can you provide the result of this query in your Easol version, please? Screen.

**Comment by allipatev** [comment]
@LonwoLonwo sure.

The current query, for Exasol versions prior 8:
```sql
/*snapshot execution*/ select
	s.SCHEMA_NAME as OBJECT_NAME,
	s.SCHEMA_OWNER AS OWNER,
CAST(NULL AS TIMESTAMP) AS created, 
	ADAPTER_SCRIPT,
	LAST_REFRESH,
	LAST_REFRESH_BY,
	ADAPTER_NOTES,
	SCHEMA_COMMENT AS OBJECT_COMMENT, s.SCHEMA_OBJECT_ID
 from
		EXA_VIRTUAL_SCHEMAS s
	INNER JOIN
		sys.EXA_SCHEMAS o
	ON
		o.schema_name = s.SCHEMA_NAME
 ORDER BY S.SCHEMA_NAME
 ;
```
The result:

image

Query to be, for Exasol versions starting from 8:

```sql
/*snapshot execution*/ select
	s.SCHEMA_NAME as OBJECT_NAME,
	s.SCHEMA_OWNER AS OWNER,
CAST(NULL AS TIMESTAMP) AS created, 
	'"' || ADAPTER_SCRIPT_SCHEMA || '"."' || ADAPTER_SCRIPT_NAME || '"' AS ADAPTER_SCRIPT,
	LAST_REFRESH,
	LAST_REFRESH_BY,
	ADAPTER_NOTES,
	SCHEMA_COMMENT AS OBJECT_COMMENT, s.SCHEMA_OBJECT_ID
 from
		EXA_VIRTUAL_SCHEMAS s
	INNER JOIN
		sys.EXA_SCHEMAS o
	ON
		o.schema_name = s.SCHEMA_NAME
 ORDER BY S.SCHEMA_NAME
 ;
```

The result

image

The main difference is in the fourth column, "ADAPTER_SCRIPT".
With double quotes it is even better.

**Comment by LonwoLonwo** [comment]
Ok, Thanks @allipatev

**Review by LonwoLonwo — APPROVED** [review]

**Comment by uslss** [comment]
verified

**Pull request merged**